internal.nix

```diff
@@ -208,8 +261,14 @@ rec {
       # if either are missing
       content = builtins.fromJSON (builtins.readFile file);
       patchDep = (name: version:
-        if lib.hasPrefix "github:" version then
-          "file://${stringToTgzPath sourceHashFunc name version}"
+        # If the dependency is of the form github:owner/repo#branch the package-lock.json contains the specific
+        # revision that the branch was pointing at at the time of npm install.
+        # The package.json itself does not contain enough information to resolve a specific dependency,
+        # because it only contains the branch name. Therefore we cannot substitute with a nix store path.
+        # If we leave the dependency unchanged, npm will try to resolve it and fail. We therefore substitute with a
+        # wildcard dependency, which will make npm look at the lockfile.
+        if lib.hasPrefix "git" version then
```

Comment:
We should use the same regex as suggested above for this match just to be sure we are consistent.